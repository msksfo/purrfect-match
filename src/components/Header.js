import React from 'react';
import '../styles/Header.css';

const Header = () => {
    return (
        <div className='header-wrappper'>
            <h1 className="header">
                <span className="header--span">Purrrr</span>-Fect Match
            </h1>
            <p className='header--tagline'>Providing loving attention and reliable service for your furrends</p>
        </div>
        
    );
};

export default Header;
