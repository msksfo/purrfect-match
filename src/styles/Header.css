.header-wrappper {
    margin-top: auto;
    margin-bottom: 2rem;
}

.header {
    font-weight: 300;
    font-size: 2rem;
    letter-spacing: 1px;
    margin-bottom: 8px;
    padding: 0 1rem;
}

.header--span {
    font-style: italic;
    letter-spacing: 3px;
    font-family: cursive;
}

.header--tagline {
    margin: 0;
    padding: 0 1rem;
    color: #dd3112;
}
