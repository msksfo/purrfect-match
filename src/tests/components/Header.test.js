import React from 'react';
import { shallow } from 'enzyme';
import "../setupTests"

import Header from '../../components/Header';


test('shoud render the header correctly', () => {
    const wrapper = shallow(<Header />)

    expect(wrapper).toMatchSnapshot();
})