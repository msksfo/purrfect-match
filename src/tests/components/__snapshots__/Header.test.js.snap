// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoud render the header correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="header--span"
        >
          Purrrr
        </span>,
        "-Fect Match",
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Purrrr",
          "className": "header--span",
        },
        "ref": null,
        "rendered": "Purrrr",
        "type": "span",
      },
      "-Fect Match",
    ],
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="header--span"
          >
            Purrrr
          </span>,
          "-Fect Match",
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Purrrr",
            "className": "header--span",
          },
          "ref": null,
          "rendered": "Purrrr",
          "type": "span",
        },
        "-Fect Match",
      ],
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
